client/src/Features/MicrosoftLearn/MicrosoftLearnFilterComponentUtils.tsx

```diff
@@ -0,0 +1,139 @@
+import { useStore } from '../../Stores/Core';
+import { useObserver } from 'mobx-react-lite';
+import React from 'react';
+import {
+  getProductsToDisplay,
+  getTypesToDisplay,
+  getLevelsToDisplay,
+  getRolesToDisplay
+} from './MicrosoftLearnFilterUtils';
+import { FilterType } from '../../Models/Learn/FilterType.model';
+import { MicrosoftLearnFilterComponent } from './MicrosoftLearnFilterComponent';
+import { FontWeights } from '@fluentui/react';
+import { IThemeOnlyProps, SimpleComponentStyles } from '../../Core/Utils/FluentUI/typings.fluent-ui';
+import { themedClassNames } from '../../Core/Utils/FluentUI';
+
+export type FilterComponentStyles = SimpleComponentStyles<
+  'root' | 'title' | 'search' | 'optionsList' | 'subOptionsList' | 'filterItem'
+>;
+
+type FilterComponentTypes = FilterType.products | FilterType.roles | FilterType.levels | FilterType.types;
+
+export const FilterComponent = (props: { type: FilterComponentTypes; name: string }) => {
+  const { filterStore, catalog } = useStore('microsoftLearnStore');
+  return useObserver(() => {
+    const displayOptions =
+      props.type === FilterType.products
+        ? getProductsToDisplay(filterStore.displayFilter[FilterType.products], catalog?.products)
```

Comment:
see if this works, to make it consistent with other lines
```suggestion
        ? getProductsToDisplay(filterStore.displayFilter[props.type], catalog?.products)
```